exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/ConvertCountToDirectScanRule.java

```diff
@@ -0,0 +1,296 @@
+
+    // if counts could not be determined, rule won't be applied
+    if (result.isEmpty()) {
+      logger.debug("Rule does not apply since one or more COUNTs could not be determined from metadata.");
+      return;
+    }
+
+    List<Path> fileList =
+            ImmutableList.of(Metadata.getSummaryFileName(formatSelection.getSelection().getSelectionRoot()));
+
+    final RelDataType scanRowType = CountToDirectScanUtils.constructDataType(agg, result.keySet());
+
+    final DynamicPojoRecordReader<Long> reader = new DynamicPojoRecordReader<>(
+        CountToDirectScanUtils.buildSchema(scanRowType.getFieldNames()),
+        Collections.singletonList((List<Long>) new ArrayList<>(result.values())));
+
+    final ScanStats scanStats = new ScanStats(ScanStats.GroupScanProperty.EXACT_ROW_COUNT, 1, 1, scanRowType.getFieldCount());
+    final MetadataDirectGroupScan directScan = new MetadataDirectGroupScan(reader, fileList, scanStats, true);
+
+    final DrillDirectScanRel newScan = new DrillDirectScanRel(scan.getCluster(), scan.getTraitSet().plus(DrillRel.DRILL_LOGICAL),
+      directScan, scanRowType);
+
+    final DrillProjectRel newProject = new DrillProjectRel(agg.getCluster(), agg.getTraitSet().plus(DrillRel.DRILL_LOGICAL),
+      newScan, CountToDirectScanUtils.prepareFieldExpressions(scanRowType), agg.getRowType());
+
+    call.transformTo(newProject);
+  }
+
+  private Pair<Boolean, Metadata_V4.MetadataSummary> checkMetadataForScanStats(DrillTable drillTable, FormatSelection formatSelection) {
+
+    // Currently only support metadata rowcount stats for Parquet tables
+    FormatPluginConfig formatConfig = formatSelection.getFormat();
+    if (!((formatConfig instanceof ParquetFormatConfig)
+      || ((formatConfig instanceof NamedFormatPluginConfig)
+      && ((NamedFormatPluginConfig) formatConfig).name.equals("parquet")))) {
+      return new ImmutablePair<Boolean, Metadata_V4.MetadataSummary>(false, null);
+    }
+
+    FileSystemPlugin plugin = (FileSystemPlugin) drillTable.getPlugin();
+    DrillFileSystem fs = null;
+    try {
+       fs = new DrillFileSystem(plugin.getFormatPlugin(formatSelection.getFormat()).getFsConf());
+    } catch (IOException e) {
+      logger.warn("Unable to create the file system object for retrieving statistics from metadata cache file ", e);
+      return new ImmutablePair<Boolean, Metadata_V4.MetadataSummary>(false, null);
+    }
+
+    // check if the cacheFileRoot has been set: this is needed because after directory pruning, the
+    // cacheFileRoot could have been changed and not be the same as the original selectionRoot
+    Path selectionRoot = formatSelection.getSelection().getCacheFileRoot() != null ?
+            formatSelection.getSelection().getCacheFileRoot() :
+            formatSelection.getSelection().getSelectionRoot();
+    Metadata_V4.MetadataSummary metadataSummary = Metadata.getSummary(fs, selectionRoot, false, ParquetReaderConfig.builder().build());
```

Comment:
Can we do this 
ParquetReaderConfig  parquetReaderConfig= ParquetReaderConfig.builder()
        .withFormatConfig((ParquetFormatConfig) formatConfig)
        .build();
Metadata.getSummary(fs, selectionRoot, false, parquetReaderConfig);

Reply:
Agree, also it is important to pass `OptionManager` into `ParquetReaderConfig`: `withOptions(settings.getOptions())` to handle some cases when statistics for varchars are allowed.

Reply:
Thanks for the suggestions.  Changed this.